xla/pjrt/pjrt_stream_executor_client.h

```diff
@@ -850,6 +924,17 @@ class PjRtStreamExecutorLoadedExecutable : public PjRtLoadedExecutable {
     if (proto != nullptr) {
       memory_stats.serialized_hlo_proto = proto->SerializeAsString();
     }
+    // TODO: Should this code live in backend-specific subclasses?
+    // (There is TfrtCpuExecutable, but no equivalent for GPUs.)
+    if (auto exec = dynamic_cast<xla::cpu::CpuExecutable*>(
```

Comment:
The `StreamExecutor` code is unused on CPU these days. The implementation for CPU is here: https://github.com/openxla/xla/blob/b3304688637b6cdcd5b016d3044b4707b06b15d7/xla/pjrt/cpu/cpu_client.h#L549

Can you update that as well? I think it just involves moving your `Populate` method into a separate module and calling it from both places.